can't run gazebo simulation on ubuntu

My ubuntu version is 16.04, and I have followed all the setup steps on px4.io. When I run the simulation with gazebo,  `make posix_sitl_default gazebo`, I got fatal error.

Here is an excerpt of the console message, where it starts to report erros. 


`/home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:198:6: error: prototype for ‘void gazebo::GimbalControllerPlugin::OnPitchStringMsg(ConstGzStringPtr&)’ does not match any in class ‘gazebo::GimbalControllerPlugin’
 void GimbalControllerPlugin::OnPitchStringMsg(ConstGzStringPtr &_msg)
      ^
In file included from /home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:20:0:
/home/haoyu/src/Firmware/Tools/sitl_gazebo/include/gazebo_gimbal_controller_plugin.hh:49:19: error: candidate is: void gazebo::GimbalControllerPlugin::OnPitchStringMsg(ConstAnyPtr&)
     private: void OnPitchStringMsg(ConstAnyPtr &_msg);
                   ^
/home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:205:6: error: prototype for ‘void gazebo::GimbalControllerPlugin::OnRollStringMsg(ConstGzStringPtr&)’ does not match any in class ‘gazebo::GimbalControllerPlugin’
 void GimbalControllerPlugin::OnRollStringMsg(ConstGzStringPtr &_msg)
      ^
In file included from /home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:20:0:
/home/haoyu/src/Firmware/Tools/sitl_gazebo/include/gazebo_gimbal_controller_plugin.hh:50:19: error: candidate is: void gazebo::GimbalControllerPlugin::OnRollStringMsg(ConstAnyPtr&)
     private: void OnRollStringMsg(ConstAnyPtr &_msg);
                   ^
/home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:212:6: error: prototype for ‘void gazebo::GimbalControllerPlugin::OnYawStringMsg(ConstGzStringPtr&)’ does not match any in class ‘gazebo::GimbalControllerPlugin’
 void GimbalControllerPlugin::OnYawStringMsg(ConstGzStringPtr &_msg)
      ^
In file included from /home/haoyu/src/Firmware/Tools/sitl_gazebo/src/gazebo_gimbal_controller_plugin.cpp:20:0:
/home/haoyu/src/Firmware/Tools/sitl_gazebo/include/gazebo_gimbal_controller_plugin.hh:51:19: error: candidate is: void gazebo::GimbalControllerPlugin::OnYawStringMsg(ConstAnyPtr&)
     private: void OnYawStringMsg(ConstAnyPtr &_msg);
                   ^
CMakeFiles/rotors_gazebo_gimbal_controller_plugin.dir/build.make:62: recipe for target 'CMakeFiles/rotors_gazebo_gimbal_controller_plugin.dir/src/gazebo_gimbal_controller_plugin.cpp.o' failed
make[7]: *** [CMakeFiles/rotors_gazebo_gimbal_controller_plugin.dir/src/gazebo_gimbal_controller_plugin.cpp.o] Error 1
CMakeFiles/Makefile2:178: recipe for target 'CMakeFiles/rotors_gazebo_gimbal_controller_plugin.dir/all' failed
make[6]: *** [CMakeFiles/rotors_gazebo_gimbal_controller_plugin.dir/all] Error 2
make[6]: *** Waiting for unfinished jobs....
[ 54%] Building CXX object CMakeFiles/rotors_gazebo_mavlink_interface.dir/src/geo_mag_declination.cpp.o
[ 57%] Linking CXX shared library libLiftDragPlugin.so
[ 57%] Built target LiftDragPlugin
[ 60%] Linking CXX shared library librotors_gazebo_controller_interface.so
[ 60%] Built target rotors_gazebo_controller_interface
[ 62%] Linking CXX shared library librotors_gazebo_mavlink_interface.so
[ 62%] Built target rotors_gazebo_mavlink_interface
Makefile:149: recipe for target 'all' failed
make[5]: *** [all] Error 2
src/firmware/posix/CMakeFiles/sitl_gazebo.dir/build.make:110: recipe for target 'external/Stamp/sitl_gazebo/sitl_gazebo-build' failed
make[4]: *** [external/Stamp/sitl_gazebo/sitl_gazebo-build] Error 2
CMakeFiles/Makefile2:9086: recipe for target 'src/firmware/posix/CMakeFiles/sitl_gazebo.dir/all' failed
make[3]: *** [src/firmware/posix/CMakeFiles/sitl_gazebo.dir/all] Error 2
CMakeFiles/Makefile2:9960: recipe for target 'src/firmware/posix/CMakeFiles/gazebo.dir/rule' failed
make[2]: *** [src/firmware/posix/CMakeFiles/gazebo.dir/rule] Error 2
Makefile:3497: recipe for target 'gazebo' failed
make[1]: *** [gazebo] Error 2
Makefile:160: recipe for target 'posix_sitl_default' failed
make: *** [posix_sitl_default] Error 2
`

--- a/src/gazebo_gimbal_controller_plugin.cpp
+++ b/src/gazebo_gimbal_controller_plugin.cpp
@@ -195,21 +195,21 @@ void GimbalControllerPlugin::Init()
 #if GAZEBO_MAJOR_VERSION >= 7 && GAZEBO_MINOR_VERSION >= 4
 /// only gazebo 7.4 and above support Any
 /////////////////////////////////////////////////
-void GimbalControllerPlugin::OnPitchStringMsg(ConstGzStringPtr &_msg)
+void GimbalControllerPlugin::OnPitchStringMsg(ConstAnyPtr &_msg)
 {
 //  gzdbg << "pitch command received " << _msg->double_value() << std::endl;
   this->pitchCommand = _msg->double_value();
 }
 
 /////////////////////////////////////////////////
-void GimbalControllerPlugin::OnRollStringMsg(ConstGzStringPtr &_msg)
+void GimbalControllerPlugin::OnRollStringMsg(ConstAnyPtr &_msg)
 {
 //  gzdbg << "roll command received " << _msg->double_value() << std::endl;
   this->rollCommand = _msg->double_value();
 }
 
 /////////////////////////////////////////////////
-void GimbalControllerPlugin::OnYawStringMsg(ConstGzStringPtr &_msg)
+void GimbalControllerPlugin::OnYawStringMsg(ConstAnyPtr &_msg)
 {
 //  gzdbg << "yaw command received " << _msg->double_value() << std::endl;
   this->yawCommand = _msg->double_value();

